src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs

```diff
@@ -499,7 +506,7 @@ protected override void ApplyDocumentRemoved(DocumentId documentId)
             var document = this.CurrentSolution.GetDocument(documentId);
             if (document?.FilePath is not null)
             {
-                _applyChangesProjectFile.RemoveDocument(document.FilePath);
+                _applyChangesProjectFile.RemoveDocumentAsync(document.FilePath, CancellationToken.None).Wait();
```

Comment:
yup.  really dislike this.  can we make thse methods async and push the non-cancellable waits up to the topmost legacy point?

Reply:
We could, but out of scope for this PR since it'd have some non-trivial public API impacts. I'll create a follow-up task to track and contain the discussion.